renku/cli/init.py

```diff
@@ -95,97 +98,191 @@ def store_directory(ctx, param, value):
+            '[{0}] {1}: {2}'.format(
+                index + 1, template['name'], template['description']
+            )
+        )
+    return '\n'.join(template_descriptions)
 
 
 @click.command()
 @click.argument(
-    'directory',
+    'path',
     default='.',
     callback=store_directory,
-    type=click.Path(writable=True, file_okay=False, resolve_path=True)
+    type=click.Path(writable=True, file_okay=False, resolve_path=True),
+)
+@click.option(
+    '--name',
+    'name',
+    default='Renku project',
+    show_default=True,
+    callback=validate_name,
+    help='Project name.',
 )
-@click.option('--name', callback=validate_name)
-@click.option('--force', is_flag=True, help='Override project templates.')
+@click.option('--template', help='Provide the name of the template to use.')
+@click.option(
+    '--template-source', help='Provide the templates repository url or path.'
+)
+@click.option(
+    '--template-ref',
+    default='master',
+    help='Specify the reference to checkout on remote template repository.',
+)
+@click.option('--description', help='Describe your project.')
+@click.option(
+    '--print-manifest', is_flag=True, help='Print templates manifest only.'
+)
+@click.option('--force', is_flag=True, help='Override target path.')
 @option_use_external_storage
 @pass_local_client
 @click.pass_context
-def init(ctx, client, directory, name, force, use_external_storage):
-    """Initialize a project."""
+def init(
+    ctx, client, use_external_storage, path, name, template, template_source,
+    template_ref, description, print_manifest, force
+):
+    """Initialize a project in PATH. Default is current path."""
+    # preparation
     if not client.use_external_storage:
         use_external_storage = False
 
     ctx.obj = client = attr.evolve(
-        client,
-        path=directory,
-        use_external_storage=use_external_storage,
+        client, path=path, use_external_storage=use_external_storage
     )
 
-    msg = 'Initialized empty project in {path}'
-    branch_name = None
-
-    stack = contextlib.ExitStack()
-
-    if force and client.repo:
-        msg = 'Initialized project in {path} (branch {branch_name})'
-        merge_args = ['--no-ff', '-s', 'recursive', '-X', 'ours']
-
+    def print_done():
+        click.echo(' DONE ', nl=False)
+        click.echo(click.style('\u2713', fg='green'))
+
+    # verify path status
+    if len(list(client.path.glob('**/*'))) > 0:
+        if not force:
+            error = FileExistsError(
+                'Folder "{0}" is not empty. Please '
+                'add --force flag to transform it into a Renku repository.'.
+                format(str(path))
+            )
+            raise click.UsageError(error)
+        else:
+            try:
+                commit = client.find_previous_commit(
+                    str(client.renku_metadata_path),
+                )
+                branch_name = 'pre_renku_init_' + str(commit)[:8]
```

Comment:
```suggestion
                branch_name = 'pre_renku_init_'.format(str(commit)[:7])
```

or 
```suggestion
                branch_name = 'pre_renku_init_'.format(commit.hexsha[:7])
```

Reply:
Going for the second.
`branch_name = 'pre_renku_init_{0}'.format(commit.hexsha[:7])`